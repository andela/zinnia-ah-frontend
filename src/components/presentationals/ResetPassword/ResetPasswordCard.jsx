import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';

import Title from '../Title/Title';
import Button from '../Button/Button';

import '../AuthenticationCard/AuthenticationCard.scss';
import '../Form.scss';

const ResetPasswordCard = ({
  password,
  handleChange,
  validationErrors,
  handleSubmit,
}) => {
  return (
    <div className="right">
      <div className="auth-card reset-password">
        <Title content="Reset Password" className="text-center" />
        <hr className="hr" />
        <Form onSubmit={handleSubmit}>
          <Form.Field>
            <input
              placeholder="New Password"
              type="password"
              onChange={handleChange}
              value={password}
              name="password"
            />
            {validationErrors.length > 0 && (
              <div className="ui negative message">
                {validationErrors.map(error => (
                  <p key={error}>{error}</p>
                ))}
              </div>
            )}
          </Form.Field>
          <Button type="submit" value="RESET PASSWORD" className="btn-dark" />
        </Form>
      </div>
    </div>
  );
};

ResetPasswordCard.propTypes = {
  password: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  validationErrors: PropTypes.array,
  handleSubmit: PropTypes.func.isRequired,
};

export default ResetPasswordCard;
