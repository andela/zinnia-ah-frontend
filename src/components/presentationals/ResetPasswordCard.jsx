import React from 'react';
import { Form } from 'semantic-ui-react';

import Title from './Title.jsx';
import Button from './Button.jsx';
import './AuthenticationCard.scss';
import './Form.scss';

const ForgotPasswordCard = () => {
  return (
    <div className="right">
      <div className="card reset-password">
        <Title title={title} />
        <hr />
        <Form>
          <Form.Field>
            <input placeholder="New Password" type="password" />
          </Form.Field>
          <Form.Field>
            <input placeholder="Confirm Password" type="password" />
          </Form.Field>
          <Button button={buttonValues} />
        </Form>
      </div>
    </div>
  );
};

const title = {
  content: 'Reset Password',
  class: 'text-center',
};

const buttonValues = {
  type: 'submit',
  value: 'RESET PASSWORD',
  class: 'btn-dark',
};

export default ForgotPasswordCard;
