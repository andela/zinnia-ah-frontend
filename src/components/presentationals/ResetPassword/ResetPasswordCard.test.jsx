import React from 'react';
import { mount } from 'enzyme';

import ResetPasswordCard from './ResetPasswordCard';

describe('ForgotPasswordCard', () => {
  it('should render correctly', () => {
    const component = mount(<ResetPasswordCard />);
    expect(component).toMatchSnapshot();
  });
});
