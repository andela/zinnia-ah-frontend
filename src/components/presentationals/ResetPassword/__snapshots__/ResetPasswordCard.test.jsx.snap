// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordCard should render correctly 1`] = `
<ForgotPasswordCard>
  <div
    className="right"
  >
    <div
      className="card reset-password"
    >
      <Title
        title={
          Object {
            "class": "text-center",
            "content": "Reset Password",
          }
        }
      >
        <h1
          className="title undefined"
        />
      </Title>
      <hr
        className="hr"
      />
      <Form
        as="form"
      >
        <form
          className="ui form"
          onSubmit={[Function]}
        >
          <FormField>
            <div
              className="field"
            >
              <input
                placeholder="New Password"
                type="password"
              />
            </div>
          </FormField>
          <FormField>
            <div
              className="field"
            >
              <input
                placeholder="Confirm Password"
                type="password"
              />
            </div>
          </FormField>
          <Button
            button={
              Object {
                "class": "btn-dark",
                "type": "submit",
                "value": "RESET PASSWORD",
              }
            }
          >
            <button
              className="btn undefined"
            />
          </Button>
        </form>
      </Form>
    </div>
  </div>
</ForgotPasswordCard>
`;
